src/acb_theta/eld_init.c

```diff
@@ -0,0 +1,25 @@
+/*
+    Copyright (C) 2023 Jean Kieffer
+
+    This file is part of Arb.
+
+    Arb is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <http://www.gnu.org/licenses/>.
+*/
+
+#include "acb_theta.h"
+
+void
+acb_theta_eld_init(acb_theta_eld_t E, slong d, slong g)
+{
+    acb_theta_eld_dim(E) = d;
+    acb_theta_eld_ambient_dim(E) = g;
+    E->last_coords = flint_malloc((g - d) * sizeof(slong));
```

Comment:
Can `g` be smaller than `d` here? Seems like a possible scenario, so I would perhaps throw in that case.